### PR TITLE
fix incremental bug in llroad test

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
1. Fix filter ddl bug. in some case that ddls has same tableName but different DB name
2. incremental restore should execute DDL before create Database.
3. `nothing to do` shouldn't return error, because even if there is no sst files, we still need to execute ddl jobs as soon as possible in restore.

### Check List 

Tests 

 - Manual test

### CODE REVIEW

File: pkg/restore/db.go [hunk 1]
```diff
@@ -210,12 +210,13 @@ func FilterDDLJobs(allDDLJobs []*model.Job, tables []*utils.Table) (ddlJobs []*m
 		tableNames[table.Info.Name.String()] = true
 		for _, job := range allDDLJobs {
 			if job.BinlogInfo.TableInfo != nil {
-				if tableIDs[job.TableID] || tableNames[job.BinlogInfo.TableInfo.Name.String()] {
+				name := job.SchemaName + job.BinlogInfo.TableInfo.Name.String()
```
Comment (kennytm) [inline pkg/restore/db.go]: 🤔 

Can we use a `struct { schemaName string; tableName string }` as the key? The current scheme will still break for `"abc" + "de"` vs `"ab" + "cde"`.

**Comment by 3pointer** [comment]
> Could you add a test?

already add this test case in llroad cases, so that we can find this error

**Comment by kennytm** [comment]
> already add this test case in llroad cases, so that we can find this error

llroad is still private, is non-trivial to setup, and is outside of this repository. If possible please still include a minimized test case.

**Comment by 3pointer** [comment]
> > already add this test case in llroad cases, so that we can find this error
> 
> llroad is still private, is non-trivial to setup, and is outside of this repository. If possible please still include a minimized test case.

ok, I'll add tests when this PR can merge

File: pkg/restore/db.go [hunk 2]
```diff
@@ -207,15 +207,17 @@ func FilterDDLJobs(allDDLJobs []*model.Job, tables []*utils.Table) (ddlJobs []*m
 		tableIDs := make(map[int64]bool)
 		tableIDs[table.Info.ID] = true
 		tableNames := make(map[string]bool)
-		tableNames[table.Info.Name.String()] = true
+		name := fmt.Sprintf("%s:%s", table.Db.Name.String(), table.Info.Name.String())
```
Comment (kennytm) [inline pkg/restore/db.go]: though less likely, there is still problem with `:` in the table/db name.

**Pull request merged**